cli/register/register.go

```diff
@@ -342,10 +342,15 @@ func DoIt(org, pattern, nodeIdTok, userPw, inputFile string, nodeOrgFromFlag str
 		cliutils.ExchangePutPost("Exchange", http.MethodPut, cliutils.GetExchangeUrl(), "orgs/"+org+"/nodes/"+nodeId+"/policy", cliutils.OrgAndCreds(org, nodeIdTok), []int{201}, exchNodePol, nil)
 	}
 
+	clusterNamesapce := ""
+	if nodeType == persistence.DEVICE_TYPE_CLUSTER {
+		clusterNamesapce = cutil.GetClusterNamespace()
+	}
+
 	// Initialize the Horizon device (node)
 	msgPrinter.Printf("Initializing the Horizon node with node type '%v'...", nodeType)
 	msgPrinter.Println()
-	nd := api.HorizonDevice{Id: &nodeId, Token: &nodeToken, Org: &org, Pattern: &pattern, Name: &nodeName, NodeType: &nodeType, HAGroup: &haGroupName}
+	nd := api.HorizonDevice{Id: &nodeId, Token: &nodeToken, Org: &org, Pattern: &pattern, Name: &nodeName, NodeType: &nodeType, ClusterNamespace: &clusterNamesapce, HAGroup: &haGroupName}
```

Comment:
No need and should not set ClusterNamespace in this call because the code in path_node.go above will discover the ClusterNamespace and save it to the exchange node. For the local GET /node API in the agent, it should always get it from cutil.GetClusterNamespace() function. Please add it here https://github.com/open-horizon/anax/blob/master/api/path_node.go#L57
Btw, ClusterNamespace is considered readonly to users. They should never need to set it up in agent and the exchange node. Agent discovers it and sets it to the exchange node.